nixos/lib/systemd-unit-options.nix

```diff
@@ -48,6 +48,18 @@ in rec {
       '';
     };
 
+    representation = mkOption {
+      default = "auto";
+      type = types.enum [ "auto" "drop-in" ];
+      description = lib.mdDoc ''
+        Defines how unit configuration is represented for systemd:
+        `auto` creates a unit file when no unit file is available
+        otherwise a drop-in.
+        `drop-in` creates a drop-in file named `overrides.conf`.
```

Comment:
Perhaps add something like 
"Use 'drop-in' for template units (units with @ in the name)" 
to make it more clear to the user when this is intended to be used?

Reply:
Good point, I added an example